src/linker/Linker.Dataflow/AttributeDataFlow.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using ILLink.Shared.TrimAnalysis;
+using Mono.Cecil;
+using Mono.Linker.Steps;
+using MultiValue = ILLink.Shared.DataFlow.ValueSet<ILLink.Shared.DataFlow.SingleValue>;
+
+namespace Mono.Linker.Dataflow
+{
+	public readonly struct AttributeDataFlow
+	{
+		readonly LinkContext _context;
+		readonly MarkStep _markStep;
+		readonly MessageOrigin _origin;
+
+		public AttributeDataFlow (LinkContext context, MarkStep markStep, in MessageOrigin origin)
```

Comment:
We only ever use one instance method once before the struct goes out of scope. I know since it's a struct on the stack, it shouldn't allocate so it isn't less performant than static methods, but is there a reason we follow this pattern instead of making the methods static?